### ISSUE TITLE
Takes up lot of memory to render Large PDF docs where as it works smoothly in Vanilla PDF.JS

### ISSUE THREAD

**Issue by vishnuys** [instructure/pdf-annotate.js#67]
All the features work perfectly fine for small documents below 100 pages the for documents that are above 300 pages makes the browser really slow that even mouse movement takes a while to reflect. The browser memory, as well as the RAM, increases at least 2GB when the document gets loaded. 

Is there any way to reduce the memory consumption or alternate ways to render the document server side?

**Comment by SAanish** [comment]
@vishnuys Hello we are facing same issue. Do you find any solution for large pdf file? Thanks

**Comment by vishnuys** [comment]
@SAanish No, I looked into the library if I can remove any unused parts of it but it seems like everything is necessary. So as a temporary hack, We are splitting the pdf into 50 page PDFs and switching it to the corresponding PDF when necessary.

**Comment by Vishnupriya112** [comment]
Am also facing the same Issue. But I tried to update the version. Its little bit faster. But I need more faster than that
